python/tank/platform/engine.py

```diff
@@ -618,6 +625,17 @@ def created_qt_dialogs(self):
         """
         return self.__created_qt_dialogs
 
+    @property
+    def emitter(self):
+        """
+        The engine's event emitter object. This object can be used by deriving
```

Comment:
trying to think through how this would work in maya or nuke - if we don't rely on QT, i guess there would be code like this:

```python
def some_maya_event_handler():
  engine = sgtk.platform.current_bundle()
  fo = sgtk.platform.events.FileOpenEvent(file_name)
  engine.emit(fo)
```

Wondering if it makes sense to have an `sgtk.platform.events` substructure where each event class is a single file - that will make the code really browseable/readable I think.